pkgs/development/python-modules/crytic-compile/default.nix

```diff
@@ -18,27 +19,31 @@ buildPythonPackage rec {
     owner = "crytic";
     repo = "crytic-compile";
     rev = "refs/tags/${version}";
-    hash = "sha256-4iTvtu2TmxvLTyWm4PV0+yV1fRLYpJHZNBgjy1MFLjM=";
+    hash = "sha256-Nx3eKy/0BLg82o3qDHjxcHXtpX3KDdnBKYwCuTLWRUE=";
   };
 
   propagatedBuildInputs = [
     cbor2
     pycryptodome
     setuptools
+    solc-select
   ];
 
   # Test require network access
   doCheck = false;
 
+  # required for import check to work
+  # PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
+  env.HOME = "/tmp";
```

Comment:
normally we do in preCheck:

```
export HOME=$(mktemp -d)
```

Reply:
I tried either 'preCheck' or 'preDist', they had no effect. I am a bit confused.

Reply:
see: pkgs/development/interpreters/python/hooks/python-imports-check-hook.sh

```
# Setup hook for checking whether Python imports succeed
echo "Sourcing python-imports-check-hook.sh"

pythonImportsCheckPhase () {
    echo "Executing pythonImportsCheckPhase"

    if [ -n "$pythonImportsCheck" ]; then
        echo "Check whether the following modules can be imported: $pythonImportsCheck"
        export PYTHONPATH="$out/@pythonSitePackages@:$PYTHONPATH"
        ( cd $out && eval "@pythonCheckInterpreter@ -c 'import os; import importlib; list(map(lambda mod: importlib.import_module(mod), os.environ[\"pythonImportsCheck\"].split()))'" )
    fi
}

if [ -z "${dontUsePythonImportsCheck-}" ]; then
    echo "Using pythonImportsCheckPhase"
    preDistPhases+=" pythonImportsCheckPhase"
fi
```

I honestly don't know how to fix it for now.

Reply:
There should be something like preDist, too but yeah, not so important and also slightly annoying that upstream is doing more than just importing when importing

Reply:
> slightly annoying that upstream is doing more than just importing when importing

Oh, that reminds me that I wanted to create a ticket in upstream pointing this out.